test/chart/AreaChart.spec.tsx

```diff
@@ -128,6 +153,225 @@ describe('AreaChart', () => {
     );
     expect(container.querySelectorAll('.recharts-area-area')).toHaveLength(1);
     expect(container.querySelectorAll('.recharts-area-curve')).toHaveLength(1);
+
+    expectAreaCurve(container, [
+      {
+        d: 'M95,5C91.25,5.667,87.5,6.333,87.5,7C87.5,7.667,87.5,8.333,87.5,9C87.5,9.667,80,10.333,80,11C80,11.667,85.85,12.333,85.85,13C85.85,13.667,82.513,14.333,79.175,15',
+      },
+    ]);
+  });
+
+  test('renders a stacked percentage chart', () => {
+    const toPercent = (decimal: number, fixed = 0) => `${(decimal * 100).toFixed(fixed)}%`;
+
+    const areaSpy = vi.fn();
+    const xAxisTicksSpy = vi.fn();
+    const Comp = (): null => {
+      const areaSettings: AreaSettings = {
+        baseValue: undefined,
+        stackId: '1',
+        dataKey: 'uv',
+        connectNulls: false,
+        data: undefined,
+      };
+      areaSpy(useAppSelector(state => selectArea(state, 0, 0, false, areaSettings)));
+      xAxisTicksSpy(useAppSelector(state => selectTicksOfAxis(state, 'xAxis', 0, false)));
+      return null;
+    };
+
+    const { container } = render(
+      <AreaChart
+        width={500}
+        height={400}
+        data={pageData}
+        stackOffset="expand"
+        margin={{
+          top: 10,
+          right: 30,
+          left: 20,
+          bottom: 20,
+        }}
+      >
+        <XAxis dataKey="name" />
+        <YAxis tickFormatter={toPercent} />
+        <Area dataKey="uv" stackId="1" />
+        <Area dataKey="pv" stackId="1" />
+        <Area dataKey="amt" stackId="1" />
+        <Customized component={<Comp />} />
+      </AreaChart>,
+    );
+
+    expect(xAxisTicksSpy).toHaveBeenLastCalledWith([
+      {
+        coordinate: 80,
+        index: 0,
+        offset: 0,
+        value: 'Page A',
+      },
+      {
+        coordinate: 145,
+        index: 1,
+        offset: 0,
+        value: 'Page B',
+      },
+      {
+        coordinate: 210,
+        index: 2,
+        offset: 0,
+        value: 'Page C',
+      },
+      {
+        coordinate: 275,
+        index: 3,
+        offset: 0,
+        value: 'Page D',
+      },
+      {
+        coordinate: 340,
+        index: 4,
+        offset: 0,
+        value: 'Page E',
+      },
+      {
+        coordinate: 405,
+        index: 5,
+        offset: 0,
+        value: 'Page F',
+      },
+      {
+        coordinate: 470,
+        index: 6,
+        offset: 0,
+        value: 'Page G',
+      },
+    ]);
+    expect(xAxisTicksSpy).toHaveBeenCalledTimes(3);
+
+    // For some reason this assertion always fails but never shows what's the difference.
```

Comment:
hm, maybe it is in a different order?

Reply:
Different property order will show in the diff. It might be some object property that doesn't get serialised.